woocommerce/wc-admin
# Update returning customer data to use customer ID

Fixes #558 

Updates the order stats table to use customer ID and provides accurate data for returning/new customers.  This PR includes changes that:
* Uses the new customer ID to associate both users and guests with an order
* Return the new/returning **customers** and not new/returning **orders** in order stats endpoint.

Some notes on how new/returning customers are calculated:
* Only orders with a `customer_id` field will be calculated.  You may need to rebuild your reports to get accurate information.
* Customers are only counted once no matter how many orders they have placed.
* If you request a period where a customer placed their first order and subsequent orders, this person is tracked as new since they became a new customer during this period.

### Before
<img width="901" alt="screen shot 2018-12-27 at 5 29 05 pm" src="https://user-images.githubusercontent.com/10561050/50474698-f34f9b80-09fc-11e9-99e5-1890c6cc811f.png">

### After
<img width="866" alt="screen shot 2018-12-27 at 5 28 46 pm" src="https://user-images.githubusercontent.com/10561050/50474695-f21e6e80-09fc-11e9-83a1-8f1d93d70295.png">

### Detailed test instructions:

1.  Make sure you have some orders with the `customer_id` field and some orders from the same user. (note that you may need to deactivate and reactivate this plugin as well as run the rebuild reports tool to get customer IDs filled in).
2.  Open the order reports page. `/wp-admin/admin.php?page=wc-admin#/analytics/orders`
3.  Check that returning and new customer stats are correct and reflect customer numbers instead of returning/new orders.
4.  Check that new + returning customers is equal to total customers during that time period.
5.  Check that a customer with multiple orders, but having their initial order in a given time period is counted as a new customer.
